### PR TITLE
Fix: support validation for map values with struct types

### PR DESCRIPTION
**Pull request by junaid2005p**
## Fixes 

Fixed support for validating struct values inside map fields. Previously, if no tag followed `endkeys`, the struct values were skipped. This PR ensures such struct values are still validated according to their own tags.

### Enhancements to validation logic:
* Updated `validator.go` to handle map values that are structs or pointers to structs by diving into their fields for validation. 
* Added test cases covering struct validation in maps, including basic rules, pointer structs, nested structs, and edge cases like nil pointers.

Check #1430 for more details.

**Make sure that you've checked the boxes below before you submit PR:**
- [x] Tests exist or have been written that cover this particular change.

@go-playground/validator-maintainers

### CODE REVIEW

**Comment by junaid2005p** [comment]
Hi @nodivbyzero, I’m new here. I’d really appreciate it if you could kindly take a look whenever you get a chance.

File: validator_test.go [hunk 1]
```diff
@@ -14335,3 +14335,317 @@ func TestValidateFn(t *testing.T) {
...
+	obj := Outer{
+		Data: map[string]Inner{
+			"key1": {Value: "exceeds"},
+		},
+	}
+	validate := New()
+	err := validate.Struct(obj)
+	if err == nil {
+		t.Fatal("Expected validation error due to Value > 5 chars")
+	}
```
Comment (nodivbyzero) [inline validator_test.go:14354]: Let's replace all 'if err == nil => t.Fatal' in new tests and add comments to keep things consistent.
```suggestion
	NotEqual(t, err, nil)
```

**Comment by nodivbyzero** [comment]
> Hi @nodivbyzero, I’m new here. I’d really appreciate it if you could kindly take a look whenever you get a chance.

Thanks for making these changes! 
Everything looks good to me - just a small suggestion regarding the asserts.

**Comment by junaid2005p** [comment]
Hey @nodivbyzero, thanks for the kind guidance. I have refactored the tests to use Equal/NotEqual to keep things consistent with the rest of the codebase and added comments as well. Let me know if I need to make any other changes.

**Review by nodivbyzero — APPROVED** [review]
